pkg/client/github_client.go

```diff
@@ -0,0 +1,26 @@
+package client
+
+import (
+	"context"
+
+	"github.com/gofri/go-github-ratelimit/github_ratelimit"
+	"github.com/google/go-github/v52/github"
+	"golang.org/x/oauth2"
+)
+
+// NewGitHubClient return a client that interacts with GitHub and has rate limiter configured.
+// With authenticated GitHub api you can make 5,000 requests per hour.
+// But the Search API has a custom rate limit.
+// Unauthenticated clients are limited to 10 requests per minute, while authenticated clients can make up to 30 requests per minute.
+// see: https://github.com/google/go-github#rate-limiting
+//
+// The RoundTripper waits for the secondary rate limit to finish in a blocking mode and then issues/retries requests.
```

Comment:
From what I can see the `RoundTripper` helps with secondary rate limiting applied by GH: https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#secondary-rate-limits

Especially I was concerned about the "concurrent requests" part of that doc, since the controllers running on different clusters might call GH in parallel. 

About :
```
In the error cases, we would run out of the quota much quickly.
```
I think you are correct, probably I could implement the delay mechanism we discussed some time ago:
- save the last time we called the GH api in memory (using a variable in the controller), and trigger further calls only when there is let's say 5 minutes delay after last call . WDYT?

Reply:
Yes, having the fixed gaps between the calls should be the safest - 1 minute should be sufficient

> Especially I was concerned about the "concurrent requests" part of that doc, since the controllers running on different clusters might call GH in parallel.

would it be still a valid concern? I know that the concurrent calls could still happen, but that would be a very rare corner-case

Reply:
> Yes, having the fixed gaps between the calls should be the safest - 1 minute should be sufficient
ok , let me work on the fixed delays between calls.

> would it be still a valid concern? I know that the concurrent calls could still happen, but that would be a very rare corner-case
In theory it could happen, it's hard to say since Github doesn't provide clear requirements about concurrent calls. 

Anyway we can:
a ) remove this `RoundTripper` library and eventually see how it behaves without it, and add it if needed.
OR
b) provide a `Single Sleep Limit` option to the `RoundTripper`, so that we know for how long it will block the controller in case it happens (see: https://github.com/gofri/go-github-ratelimit#options)

Reply:
I've added the delay checks in host operator , please check https://github.com/codeready-toolchain/host-operator/pull/819/commits/d0384c77815b726c068230f89c018fdce41f3c5e

I'll eventually do the same in member-operator, once we clear out all the concerns in host-operator.

Reply:
@MatousJobanek I've also removed the GH rate limiter , as discussed earlier today, please see https://github.com/codeready-toolchain/toolchain-common/pull/315/commits/96782e36fc95cac6fb253aa8a5b9a7c9c52ddbb6 
and updated host-operator PR as accordingly: https://github.com/codeready-toolchain/host-operator/pull/819/commits/3b346c76d927a66755a277eb590c2acce0056754